bftengine/include/bftengine/KeyExchangeManager.hpp

```diff
@@ -180,6 +180,7 @@ class KeyExchangeManager {
   std::string generateCid(std::string);
   // build cryptosystem
   void notifyRegistry();
+  void exchangeTlsKeys(const std::string& type, uint64_t bftSn);
```

Comment:
An `enum` or `variant` might make more sense than `string` or even `bool` if the type is binary.

Reply:
The type (either server or client) is part of the final file path. This is why we move it here directly through a string.
This way, in the future, it will be easier to change the file path directly from the caller function

Reply:
I don't think it will be easier. If you want to change it then you now have to change it in multiple places. Everywhere, where `exchangeTlsKeys` is invoked and then in the function itself which checks the type.
Instead, if you care about changing the path only then you could construct the type string inside the function instead of relying on the caller.
Meaning, I suggest to introduce an enum that specifies the type and within `exchangeTlsKeys` you construct the path based on that enum. If the path ever changes then you have to change it inside `exchangeTlsKeys` only.